demo/index.html

```diff
@@ -38,6 +38,7 @@
   selector.addEventListener('change', function() {
     document.documentElement.lang = this.value;
     I18nMsg.lang = this.value;
+    Platform.performMicrotaskCheckpoint();
```

Comment:
Yes, it doesn't work on FF without this change.

```js
     var observerLang = new PathObserver(window.I18nMsg, 'lang');
      observerLang.open(function(newValue, oldValue) {
        _numInstancesUpdated = 0;
        this._setLanguage(newValue);
      }.bind(this));
```

Object.observe is being used to watch changes on I18nMsg.lang, which is changed when another  language is selected from the select box.

Maybe another option would be to make **this.setLanguage()** public?